core/src/test/java/org/bitcoinj/core/WalletTest.java

```diff
@@ -851,6 +852,418 @@ public void onWalletChanged(Wallet wallet) {
     }
 
     @Test
+    public void doubleSpendWeCreate() throws Exception {
+        // Test we keep pending double spends in IN_CONFLICT until one of them is included in a block
+        // and we handle reorgs and dependency chains properly.
+        // The following graph shows the txns we use in this test and how they are related 
+        // (Eg txA1 spends txARoot outputs, txC1 spends txA1 and txB1 outputs, etc).
+        // txARoot -> txA1 -+
+        //                  |--> txC1 -> txD1
+        // txBRoot -> txB1 -+
+        //
+        // txARoot -> txA2 -+
+        //                  |--> txC2 -> txD2
+        // txBRoot -> txB2 -+
+        //
+        // txARoot -> txA3
+        //
+        // txA1 is in conflict with txA2 and txA3. txB1 is in conflict with txB2. 
+        
+
+        CoinSelector originalCoinSelector = wallet.getCoinSelector();        
+        try {
+            wallet.setCoinSelector(AllowUnconfirmedCoinSelector.get());
```

Comment:
There is an alias for this: `wallet.allowSpendingUnconfirmedTransactions()` which is a bit clearer